File: src/leetcodejava/array/RemoveDuplicatesSortedArray26.java
package leetcodejava.array;

import org.junit.Assert;
import org.junit.Test;

/**
 * This is the solution of No. 26 problem in the LeetCode,
 * the website of the problem is as follow:
 * https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/
 * <p>
 * The description of problem is as follow:
 * ==========================================================================================================
 * 给定一个排序数组，你需要在 原地 删除重复出现的元素，使得每个元素只出现一次，返回移除后数组的新长度。
 * <p>
 * 不要使用额外的数组空间，你必须在 原地 修改输入数组 并在使用 O(1) 额外空间的条件下完成。
 * <p>
 * 示例 1:
 * <p>
 * 给定数组 nums = [1,1,2],
 * 函数应该返回新的长度 2, 并且原数组 nums 的前两个元素被修改为 1, 2。
 * 你不需要考虑数组中超出新长度后面的元素。
 * 示例 2:
 * <p>
 * 给定 nums = [0,0,1,1,1,2,2,3,3,4],
 * 函数应该返回新的长度 5, 并且原数组 nums 的前五个元素被修改为 0, 1, 2, 3, 4。
 * 你不需要考虑数组中超出新长度后面的元素。
 * <p>
 * 来源：力扣（LeetCode）
 * ==========================================================================================================
 *
 * @author zhangyu (zhangyuyu417@gmail.com)
 */
public class RemoveDuplicatesSortedArray26 {
    @Test
    public void removeDuplicatesSortedArrayTest() {
        int[] nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        int length = getLength(nums);
        Assert.assertEquals(length, 5);
    }

    /**
     * 获取数组的长度
     *
     * @param nums 有序数组
     * @return 长度
     */
    private int getLength(int[] nums) {
        int count = 0;
        for (int j = 1; j < nums.length; j++) {
            if (nums[count] != nums[j]) {
                nums[++count] = nums[j];
            }
        }
        return count + 1;
    }
}

File: src/leetcodepython/array/remove_duplicates_sorted_array_26.py
# encoding='utf-8'

'''
/**
 * This is the solution of No. 258 problem in the LeetCode,
 * the website of the problem is as follow:
 * https://leetcode-cn.com/problems/add-digits/
 *
 * The description of problem is as follow:
 * ==========================================================================================================
 * 给定一个非负整数 num，反复将各个位上的数字相加，直到结果为一位数。
 *
 * 示例:
 * 输入: 38
 * 输出: 2
 * 解释: 各位相加的过程为：3 + 8 = 11, 1 + 1 = 2。 由于 2 是一位数，所以返回 2。
 *
 * 来源：力扣（LeetCode）
 ** ==========================================================================================================
 *
 * @author zhangyu (zhangyuyu417@gmail.com)
 */
'''
from typing import List


class Solution:
    def remove_duplicates(self, nums: List[int]) -> int:
        '''
            移除重复的数字
        Args:
            nums: 数组
        Returns:
            重复数
        '''
        if nums == None or len(nums) < 1:
            return 0
        idx = 0
        for i in range(1, len(nums)):
            if nums[idx] != nums[i]:
                idx += 1
                nums[idx] = nums[i]
        return idx + 1


if __name__ == '__main__':
    nums = [1, 1, 2]
    solution = Solution()
    result = solution.remove_duplicates(nums)
    print(result)
    assert result == 2
